### PR TITLE
Better document serialization functions.

### PR DESCRIPTION
**Pull request by bangerth**
Specifically, equip all of them with a link to the BOOST library.

This is a really boring patch. @drwells would be proud of me :-)

/rebuild

### CODE REVIEW

File: include/deal.II/base/table.h [hunk 1]
```diff
@@ -618,7 +618,8 @@ class TableBase : public Subscriptor
 
   /**
    * Write or read the data of this object to or from a stream for the purpose
-   * of serialization.
+   * of serialization using the [BOOST serialization
+   * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html)..
```
Comment (masterleinad) [inline include/deal.II/base/table.h:622]:
```suggestion
   * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
```

File: include/deal.II/differentiation/sd/symengine_number_types.h [hunk 2]
```diff
@@ -480,7 +482,8 @@ namespace Differentiation
 #  ifdef DOXYGEN
       /**
        * Write and read the data of this object from a stream for the purpose
-       * of serialization.
+       * of serialization using the [BOOST serialization
+       * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html)..
```
Comment (masterleinad) [inline include/deal.II/differentiation/sd/symengine_number_types.h:486]:
```suggestion
       * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
```

File: include/deal.II/particles/particle.h [hunk 3]
```diff
@@ -421,18 +421,21 @@ namespace Particles
 
     /**
      * Write the data of this object to a stream for the purpose of
-     * serialization.
+     * serialization using the [BOOST serialization
+     * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html)..
```
Comment (masterleinad) [inline include/deal.II/particles/particle.h:425]:
```suggestion
     * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
```

File: include/deal.II/particles/particle_handler.h [hunk 4]
```diff
@@ -737,7 +737,8 @@ namespace Particles
     register_load_callback_function(const bool serialization);
 
     /**
-     * Serialize the contents of this class.
+     * Serialize the contents of this class using the [BOOST serialization
+     * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html)..
```
Comment (masterleinad) [inline include/deal.II/particles/particle_handler.h:741]:
```suggestion
     * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
```

**Comment by bangerth** [comment]
All suggested changes applied!

**Review by masterleinad — APPROVED** [review]
LGTM.

**Review by Rombur — APPROVED** [review]

**Pull request merged**